specification/confidentialledger/Microsoft.CodeTransparency/main.tsp

```diff
@@ -28,10 +27,764 @@ namespace Microsoft.CodeTransparency;
+  @doc("Service certificate used in TLS and for receipt verification")
+  serviceCertificate: string;
+
+  @doc("Service identifier")
+  serviceId: string;
+
+  @doc("Signature algorithm used in receipts")
+  signatureAlgorithm: string;
+
+  @doc("Tree algorithm used in receipts")
+  treeAlgorithm: string;
+}
+
+@removed(Versions.`2025-01-31-preview`)
+@doc("Response of entry submission containing the operationId.")
+model CreateEntryResult {
+  @doc("String representing the operation id submitted")
+  @pattern("^[0-9]+\\.[0-9]+$")
+  operationId: string;
+}
+
+@added(Versions.`2025-01-31-preview`)
+@doc("Response of entry submission if the response can be served immediately, see IETF SCITT draft")
+model CreateEntryCreated {
+  @doc("Status code")
+  @statusCode
+  statusCode: 201;
+
+  @doc("The MIME content type a Cose body is application/cose, containing a CoseSign1 signature.")
+  @header("Content-Type")
+  contentType: "application/cose";
+
+  @doc("Location of the transparent statement")
+  @header("Location")
+  location: string;
+
+  @doc("Transparent statement")
+  @bodyRoot
+  body: bytes;
+}
+
+@added(Versions.`2025-01-31-preview`)
+@doc("Response of entry submission containing the transparent statement id, see IETF SCITT draft")
+model CreateEntryPending {
+  @doc("Status code")
+  @statusCode
+  statusCode: 202;
+
+  @doc("Response content is CBOR")
+  @header("Content-Type")
+  contentType: "application/cbor";
+
+  @doc("Location of the transparent statement")
+  @header("Location")
+  location: string;
+
+  @doc("Response contains the identifier")
+  @bodyRoot
+  body: bytes;
+}
+
+@doc("Signed statement")
+model SignedStatement {
+  @doc("The MIME content type a Cose body is application/cose, containing a CoseSign1 signature, see IETF SCITT draft")
+  @header("Content-Type")
+  contentType: "application/cose";
+
+  @doc("CoseSign1 signature envelope")
+  @bodyRoot
+  body: bytes;
+}
+
+@doc("Transparent statement")
+model TransparentStatement {
+  @doc("The MIME content type a Cose body is application/cose, containing a CoseSign1 signature, see IETF SCITT draft")
+  @header("Content-Type")
+  contentType: "application/cose";
+
+  @doc("CoseSign1 signature envelope with the receipt embedded in the unprotected header")
+  @bodyRoot
+  body: bytes;
+}
+
+@doc("A ledger receipt, see IETF SCITT draft")
+model ReceiptEntry {
+  @doc("The MIME content type for receipt is application/cbor.")
+  @removed(Versions.`2025-01-31-preview`)
+  @header("Content-Type")
+  contentTypeCbor: "application/cbor";
+
+  @doc("The MIME content type for receipt is application/cose.")
+  @added(Versions.`2025-01-31-preview`)
+  @header("Content-Type")
+  contentTypeCose: "application/cose";
+
+  @doc("A raw CBOR content")
+  @bodyRoot
+  body: bytes;
+}
+
+@doc("A List of Entry IDs")
+@pagedResult
+model EntryIds {
+  @doc("List of entry ids")
+  @items
+  transactionIds: string[];
+
+  @doc("Path to the next page of results")
+  @nextLink
```

Comment:
nit: getting this error while compiling it locally: 

` error ambiguous-symbol: "nextLink" is an ambiguous name between TypeSpec.nextLink, Azure.Core.nextLink. Try using fully qualified name instead: TypeSpec.nextLink, Azure.Core.nextLink`

Reply:
Using   _@Azure.Core.nextLink_ seems to work

Reply:
Had to use `@TypeSpec.nextLink` because  `@Azure.Core.nextLink` does not compile in typespec